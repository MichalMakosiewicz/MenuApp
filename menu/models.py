from django.db import models
import os
from uuid import uuid4
from django.utils.deconstruct import deconstructible


@deconstructible
class PathAndRename(object):

    def __init__(self, sub_path):
        self.path = sub_path

    def __call__(self, instance, filename):
        ext = filename.split('.')[-1]
        filename = '{}.{}'.format(uuid4().hex, ext)
        return os.path.join(self.path, filename)


class Dish(models.Model):
    name = models.CharField(max_length=50)
    description = models.CharField(max_length=300)
    price = models.IntegerField()
    preparation_time = models.IntegerField()
    add_date = models.DateField()
    update_date = models.DateField()
    photo = models.ImageField(upload_to=PathAndRename("static"), default="Empty")
    is_vegan = models.BooleanField()

    def __str__(self):
        return self.name


class MenuCard(models.Model):
    title = models.CharField(max_length=50)
    create_date = models.DateField()
    update_date = models.DateField()
    dish = models.ManyToManyField(Dish)

    def __str__(self):
        return self.title
