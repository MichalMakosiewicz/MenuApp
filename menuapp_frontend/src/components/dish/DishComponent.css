.root {
    margin: 10px;
}

.title {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.vegan-text {
    color: green;
    margin-left: 30px;
}

.create-update {
    font-size: 10px;
    color: gray;
}

.price {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    justify-content: space-between;
}