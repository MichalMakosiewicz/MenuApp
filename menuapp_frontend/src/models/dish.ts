export interface DishInterfaceRaw {
    id: number;
    name: string;
    description: string;
    price: number;
    preparation_time: number;
    add_date: string;
    update_date: string;
    photo: string;
    is_vegan: boolean;
}

export class DishModel {
    public id: number;
    public name: string;
    public description: string;
    public price: number;
    public preparation_time: number;
    public add_date: string;
    public update_date: string;
    public photo: string;
    public is_vegan: boolean;

    constructor(dishInterface: DishInterfaceRaw) {
        this.id = dishInterface.id;
        this.name = dishInterface.name;
        this.description = dishInterface.description;
        this.price = dishInterface.price;
        this.preparation_time = dishInterface.preparation_time;
        this.add_date = dishInterface.add_date;
        this.update_date = dishInterface.update_date;
        this.photo = dishInterface.photo;
        this.is_vegan = dishInterface.is_vegan;
    }
}