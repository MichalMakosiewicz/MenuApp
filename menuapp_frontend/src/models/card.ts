import { DishModel } from './dish';

export interface CardInterfaceRaw {
    id: number;
    title: string;
    create_date: string;
    update_date: string;
    dish: number[];
}

export class CardModel {
    public id: number;
    public title: string;
    public create_date: string;
    public update_date: string;
    public dish: number[];

    constructor(cardInterface: CardInterfaceRaw) {
        this.id = cardInterface.id;
        this.title = cardInterface.title;
        this.create_date = cardInterface.create_date;
        this.update_date = cardInterface.update_date;
        this.dish = cardInterface.dish;
    }
}