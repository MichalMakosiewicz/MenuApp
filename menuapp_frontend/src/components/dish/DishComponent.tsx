import React, {useEffect} from 'react';
import {DishModel} from '../../models/dish';
import {CardModel} from '../../models/card';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import "./DishComponent.css"


interface DishComponentInterface {
    dish: DishModel;
}

export function DishComponent(props: DishComponentInterface) {

    const {dish} = props;

    useEffect(() => {
        console.log(dish);
    }, [])

    const isVegan = () => {
        if (dish.is_vegan) {
            return "VEGAN";
        } else {
            return "";
        }
    }

    return (
        <div>
            <Card className='root'>
                <CardContent>
                    <div className="title">
                        <Typography className="title" variant="h5" gutterBottom>
                            {dish.name}
                        </Typography>
                        <div className="vegan-text">
                            {isVegan()}
                        </div>
                    </div>
                    <Typography component="h1">
                        {dish.description}
                    </Typography>
                    <img src={dish.photo}/>
                    <div className="create-update">
                        <Typography>
                            creation date: {dish.add_date}
                        </Typography>
                        <Typography>
                            update date: {dish.update_date}
                        </Typography>
                    </div>
                    <div className="price">
                        <Typography>
                            preparation time: {dish.preparation_time} min
                        </Typography>
                        <Typography variant="h5">
                            {dish.price},-
                        </Typography>
                    </div>
                </CardContent>
            </Card>

        </div>
    )
}