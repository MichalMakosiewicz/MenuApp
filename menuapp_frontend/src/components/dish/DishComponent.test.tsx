import React from 'react';
import {DishComponent} from './DishComponent';
import renderer from 'react-test-renderer';
import {DishModel} from '../../models/dish';

describe('Navbar test', () => {
    const dish = new DishModel({
        id: 1,
        name: "test",
        description: "test",
        price: 12,
        preparation_time: 13,
        add_date: "test",
        update_date: "test",
        photo: "test",
        is_vegan: true
    })
    it('match the snapshot', () => {
        const tree = renderer.create(<DishComponent dish={dish}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})