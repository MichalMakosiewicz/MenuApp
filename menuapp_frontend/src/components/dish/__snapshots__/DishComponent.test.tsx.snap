// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test match the snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiCard-root root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="title"
      >
        <h5
          className="MuiTypography-root title MuiTypography-h5 MuiTypography-gutterBottom"
        >
          test
        </h5>
        <div
          className="vegan-text"
        >
          VEGAN
        </div>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-body1"
      >
        test
      </h1>
      <img
        src="test"
      />
      <div
        className="create-update"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          creation date: 
          test
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          update date: 
          test
        </p>
      </div>
      <div
        className="price"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          preparation time: 
          13
           min
        </p>
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          12
          ,-
        </h5>
      </div>
    </div>
  </div>
</div>
`;
